include/openthread/mdns.h

```diff
@@ -862,6 +928,30 @@ otError otMdnsGetNextIp4AddressResolver(otInstance            *aInstance,
                                         otMdnsAddressResolver *aResolver,
                                         otMdnsCacheInfo       *aInfo);
 
+/**
+ * Iterates over record querier entries.
+ *
+ * Requires `OPENTHREAD_CONFIG_MULTICAST_DNS_ENTRY_ITERATION_API_ENABLE`.
+ *
+ * On success, @p aQuerier is populated with information about the next querier . The `mCallback` field is always
+ * set to `NULL` as there may be multiple active querier with different callbacks. Other pointers within the
+ * `otMdnsRecordQuerier` structure remain valid until the next call to any OpenThread stack's public or platform
+ * API/callback.
+ *
+ * @param[in]  aInstance   The OpenThread instance.
+ * @param[in]  aIterator   Pointer to the iterator.
+ * @param[out] aQuerier    Pointer to an `otMdnsRecordQuerier` to return the information about the next one.
+ * @param[out] aInfo       Pointer to an `otMdnsCacheInfo` to return additional information.
+ *
+ * @retval OT_ERROR_NONE         @p aQuerier, @p aInfo, & @p aIterator are updated successfully.
+ * @retval OT_ERROR_NOT_FOUND    Reached the end of the list.
+ * @retval OT_ERROR_INVALID_ARG  @p aIterator is not valid.
+ */
+otError otMdnsGetNextRecordQuerier(otInstance          *aInstance,
```

Comment:
Just curious: Is this API only intended for cli command? Are there any other usages for it?

Reply:
All APIs have corresponding CLI commands (including this one).

This (along with other `GetBrowser/Resolver`) APIs are mainly intended for testing, but there may be other use cases. Basically, we can use them to get the full list of browsers/resolvers/queries that mDNS is currently tracking (whether it has anything in cache for them, etc.).

Support for these APIs can be controlled using `OPENTHREAD_CONFIG_MULTICAST_DNS_ENTRY_ITERATION_API_ENABLE`. So, if they are not needed/used (and we want to reduce overall code size), we can use the above config to disable them (have all the related code excluded).